File: src/main/java/mediathek/gui/dialog/reset/ResetSettingsDialog.java
package mediathek.gui.dialog.reset;

import mediathek.config.Daten;
import mediathek.gui.dialog.StandardCloseDialog;

import javax.swing.*;
import java.awt.*;

@SuppressWarnings("serial")
public class ResetSettingsDialog extends StandardCloseDialog {
    private final Daten daten;

    public ResetSettingsDialog(Frame owner, Daten daten) {
        super(owner, "Programm zurücksetzen", true);
        this.daten = daten;
        setResizable(false);

        pack();
    }

    @Override
    public JComponent createContentPanel() {
        return new ResetSettingsPanel(null, daten);
    }
}

File: src/main/kotlin/mediathek/gui/dialog/reset/ResetSettingsDialog.kt
package mediathek.gui.dialog.reset

import mediathek.daten.DatenPset
import mediathek.daten.ProgramSetRepository
import mediathek.gui.dialog.StandardCloseDialog
import mediathek.mainwindow.SettingsResetHost
import java.util.function.BiConsumer
import javax.swing.JComponent

class ResetSettingsDialog(
    private val host: SettingsResetHost,
    private val programSets: ProgramSetRepository,
    private val programSetExporter: BiConsumer<Array<DatenPset>, String>,
) : StandardCloseDialog(host.ownerFrame(), "Programm zurücksetzen", true) {
    init {
        isResizable = false
    }

    override fun createContentPanel(): JComponent = ResetSettingsPanel(host, programSets, programSetExporter)
}
